### DISCUSSION TITLE
Electrodes shape doesn't look like an ellipse

### DISCUSSION THREAD

**Discussion by Raichuu41** [simnibs/simnibs#202] [8. Simulation workflows]
Hello! I am trying to place ellipse-shaped electrodes via Python into my mesh via SimNIBS but the shape looks different:
![image](https://github.com/simnibs/simnibs/assets/25647046/ec28ef9f-bf4e-4ae4-addc-abad3eaef33d)
Why are there "triangle holes" and also triangles that are not connected to the rest of the electrode? How can I fix these issues?

This is how I add electrodes to the mesh:
```python
# globals.py
# numbers in centimeters
BOX_WIDTH = 4.5e-2
BOX_LENGTH = 4.5e-2
BOX_HEIGHT = 1e-2
CABLE_THICKNESS = 150e-6  # 150 micrometers
ELECTRODE_RADIUS = 0.5e-2
ELECTRODE_THICKNESS = 150e-6  # 150 micrometers

# main.py
# Add a TDSC list to the session
tdcs_list = SESSION.add_tdcslist()
# Electrodes with currents
electrode_positions = [
    # first electrode
    (BOX_WIDTH / 4, BOX_LENGTH / 2, BOX_HEIGHT - ELECTRODE_THICKNESS),
    # second electrode
    (3 * BOX_WIDTH / 4, BOX_LENGTH / 2, BOX_HEIGHT - ELECTRODE_THICKNESS),
]
for i, (x, y, z) in enumerate(electrode_positions):
    electrode = tdcs_list.add_electrode()
    electrode.centre = [x, y, z]
    electrode.shape = "ellipse"
    electrode.dimensions = [ELECTRODE_RADIUS, ELECTRODE_RADIUS]
    electrode.thickness = ELECTRODE_THICKNESS
    electrode.channelnr = i + 1
```

**Comment by bjornsig** [comment]
Can you provide the mesh with electrons on so we can look at it?

**Reply by Raichuu41** [reply]
Sure, no problem. I had to upload it on Google Drive because the files themselves are not supported and the zip file is too big with 196MB. Here's the link: [Mesh Files in Google Drive](https://drive.google.com/file/d/1Egb9WqQfoY9NCg6jhkhKT4IgvEPgwLpD/view?usp=sharing)
The `simulation.msh` is the base mesh. The code from above is used to add the electrodes on that mesh. And the other two files are the simnibs output where you can see that the electrodes are weirdly shaped.

**Reply by axthi** [reply]
hi, the model is very small (0.045 mm size), and I guess that this is what causes the problems. I suggest to increase by a factor of 1000, and try again.

**Reply by Raichuu41** [reply]
But it needs to be simulated in this small scale because that's the size of the original in real life, too. Do you think SimNIBS can't handle the small scale? I thought it shouldn't matter as long as the resolution of the mesh (the density of triangles) is enough/the same as a higher sized model with similar resolution.

**Reply by TorgeW** [reply]
You can scale all your coordinates up at the start, run the placement and than scale it down by the same factor again. Hope that helps :)
